packages/application-generic/src/usecases/conditions-filter/conditions-filter.usecase.ts

```diff
@@ -105,42 +504,197 @@ export class ConditionsFilter extends Filter {
     filter: StepFilter,
     variables: IFilterVariables,
+    command: ConditionsFilterCommand,
     filterProcessingDetails: FilterProcessingDetails
   ): Promise<boolean> {
-    const foundFilter = await this.findAsync(filter.children, (i) =>
-      this.processFilter(variables, i, filterProcessingDetails)
+    const { webhookFilters, otherFilters } = this.splitFilters(filter);
+
+    const foundFilter = await this.findAsync(otherFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
     );
+    if (foundFilter) {
+      return true;
+    }
 
-    return !!foundFilter;
+    return !!(await this.findAsync(webhookFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
+    ));
   }
 
-  private async processFilter(
-    variables: IFilterVariables,
-    child: FilterParts,
-    filterProcessingDetails: FilterProcessingDetails
-  ): Promise<boolean> {
-    let passed = false;
+  @Instrument()
+  private async normalizeVariablesData(command: ConditionsFilterCommand) {
+    const filterVariables: IFilterVariables = {};
 
-    if (child.on === FilterPartTypeEnum.TENANT) {
-      passed = this.processFilterEquality(
-        variables,
-        child,
-        filterProcessingDetails
+    const combinedFilters = [
+      command.step,
+      ...(command.step?.variants || []),
+    ].flatMap((variant) => (variant?.filters ? variant?.filters : []));
+
+    filterVariables.subscriber = await this.fetchSubscriberIfMissing(
+      command,
+      combinedFilters
+    );
+    filterVariables.tenant = await this.fetchTenantIfMissing(
+      command,
+      combinedFilters
+    );
+    filterVariables.payload = command.variables?.payload
+      ? command.variables?.payload
+      : command.job?.payload ?? undefined;
+
+    filterVariables.step = command.variables?.step ?? undefined;
+    filterVariables.actor = command.variables?.actor ?? undefined;
+
+    return filterVariables;
+  }
+
+  private async fetchSubscriberIfMissing(
+    command: ConditionsFilterCommand,
+    filters: IMessageFilter[]
+  ): Promise<SubscriberEntity | undefined> {
+    if (command.variables?.subscriber) {
+      return command.variables.subscriber;
+    }
+
+    const subscriberFilterExist = filters?.find((filter) => {
+      return filter?.children?.find((item) => item?.on === 'subscriber');
+    });
+
+    if (subscriberFilterExist && command.job) {
+      return (
+        (await this.getSubscriberBySubscriberId({
+          subscriberId: command.job.subscriberId,
+          _environmentId: command.environmentId,
+        })) ?? undefined
       );
     }
 
-    return passed;
+    return undefined;
+  }
+
+  private async fetchTenantIfMissing(
+    command: ConditionsFilterCommand,
+    filters: IMessageFilter[]
+  ): Promise<TenantEntity | undefined> {
+    if (command.variables?.tenant) {
+      return command.variables.tenant;
+    }
+
+    const tenantIdentifier = command.job?.tenant?.identifier;
```

Comment:
can we add here condition in case `command.job?.tenant` is the identifier? 

```suggestion
    const tenantIdentifier = typeof command.job?.tenant === "string" ? command.job?.tenant : command.job?.tenant?.identifier;
```